internal/controller/promotion/pullrequest.go

```diff
@@ -76,15 +77,17 @@ func newGitProvider(
 	var err error
 	if pullRequest.GitHub != nil {
 		gpClient, err = gitprovider.NewGitProviderServiceFromName(github.GitProviderServiceName)
+	} else if pullRequest.GitLab != nil {
+		gpClient, err = gitprovider.NewGitProviderServiceFromName(gitlab.GitProviderServiceName)
 	} else {
 		gpClient, err = gitprovider.NewGitProviderServiceFromURL(url)
 	}
```

Comment:
Nit: this is likely to become a growing list as we add support for other Git providers. Given this, it would probably be good to make this a `switch` to ease adding new providers.

```suggestion
	switch {
	case pullRequest.GitHub != nil:
		gpClient, err = gitprovider.NewGitProviderServiceFromName(github.GitProviderServiceName)
	case pullRequest.GitLab != nil:
		gpClient, err = gitprovider.NewGitProviderServiceFromName(gitlab.GitProviderServiceName)
	default:
		gpClient, err = gitprovider.NewGitProviderServiceFromURL(url)
	}
```